real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/nodes/baseNodes/PeriodicNode.java

```diff
@@ -0,0 +1,44 @@
+package com.roboclub.robobuggy.nodes.baseNodes;
+
+import java.util.Timer;
+import java.util.TimerTask;
+
+/**
+ * This node will cause for an update function to be run at a fixed frequency
+ *
+ * @author Zachary Dawson
+ *
+ */
+public abstract class PeriodicNode extends BuggyDecoratorNode {
+
+    private final int runPeriod;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Perhaps 'runPeriod' could be replaced by a local variable.](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1870702265/issues/source?bid=2938665&fileBranchId=3087231#l14)